src/refract/JSONSchemaVisitor.cc

```diff
@@ -232,30 +300,42 @@ namespace refract
              ++it) {
 
             if (*it) {
+                MemberElement *mr = TypeQueryVisitor::as<MemberElement>(*it);
+                StringElement *str = TypeQueryVisitor::as<StringElement>(mr->value.first);
+
                 if (IsTypeAttribute(*(*it), "required") ||
                     IsTypeAttribute(*(*it), "fixed") ||
                     (fixed && !IsTypeAttribute(*(*it), "optional"))) {
 
-                    MemberElement *mr = TypeQueryVisitor::as<MemberElement>(*it);
-                    StringElement *n = TypeQueryVisitor::as<StringElement>(mr->value.first);
-                    reqVals.push_back(IElement::Create(n->value));
+                    reqVals.push_back(IElement::Create(str->value));
                 }
 
-                JSONSchemaVisitor renderer(fixed);
-                renderer.visit(*(*it));
-
-                ObjectElement *o1 = TypeQueryVisitor::as<ObjectElement>(renderer.get());
-                if (!o1->value.empty()) {
-                    MemberElement *m1 = TypeQueryVisitor::as<MemberElement>(o1->value[0]->clone());
-
-                    m1->renderType(IElement::rCompact);
-                    o->push_back(m1);
+                if (IsVariableProperty(*mr->value.first)) {
+                    varProps.push_back(mr);
+                }
+                else {
+                    JSONSchemaVisitor renderer(pDefs, fixed);
+                    renderer.visit(*(*it));
+                    ObjectElement *o1 = TypeQueryVisitor::as<ObjectElement>(renderer.get());
+                    if (!o1->value.empty()) {
+                        MemberElement *m1 = TypeQueryVisitor::as<MemberElement>(o1->value[0]->clone());
```

Comment:
sometime you use `value[0]` sometime `front()`, please choose one of them.

If `cloned` element not will be of type `MemeberElement` there will be leak memory.